microprofile/tests/arquillian/src/main/java/io/helidon/microprofile/arquillian/HelidonDeployableContainer.java

```diff
@@ -108,12 +106,25 @@ public void setup(HelidonContainerConfiguration configuration) {
 
     @Override
     public void start() {
-        dummyServer = Server.builder().build();
+        try {
+            if (!CDI.current()
+                    .getBeanManager()
+                    .getExtension(ServerCdiExtension.class)
+                    .started()) {
+                dummyServer = Server.builder().build();
+            }
+        } catch (IllegalStateException e) {
+            // CDI not running
+            dummyServer = Server.builder().build();
+        }
     }
 
     @Override
     public void stop() {
         // No-op
+        if (null != dummyServer) {
+            dummyServer.stop();
+        }
     }
```

Comment:
Not a noop any more

Reply:
Fixed (and a few other code fixes in this class).